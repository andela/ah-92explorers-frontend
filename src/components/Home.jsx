import React, { Component } from 'react';
import { connect } from 'react-redux';
import { getHome } from '../redux/actions/actionCreators';

class Home extends Component {
  componentDidMount() {
    this.props.getHome();
  }

  render() {
    const { name } = this.props;
    return (
      <div>
        <h1>{ name }</h1>
      </div>
    );
  }
}

const stateNow = state => ({
  name: state.home.home,
});

export default connect(stateNow, { getHome })(Home);
