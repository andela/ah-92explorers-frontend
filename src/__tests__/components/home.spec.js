import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import Home from '../../components/Home';


const initialState = {};

// here it is possible to pass in any middleware if needed into //configureStore
const mockStore = configureStore();
let wrapper;
let store;


const props = {
  login: {},
};

describe('<Home />', () => {
  beforeEach(() => {
    // creates the store with any initial state or middleware needed
    store = mockStore(initialState);
    wrapper = shallow(<Home store={store} {...props} />);
  });

  it('should render Login component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
