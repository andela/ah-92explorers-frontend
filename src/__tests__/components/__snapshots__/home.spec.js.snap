// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render Login component 1`] = `ShallowWrapper {}`;
